src/book.py

```diff
@@ -30,6 +30,7 @@ class Book:
 
     def __init__(self) -> None:
         self.operations: list[tr.Operation] = []
+        self.current_file_path = None
```

Comment:
I think, this does not work. The Book keeps record of all operations from all files. I guess we have to save the file path to the Operations class (like the line number).

Reply:
I am currently thinking of something like.. but this could be overkill. I am unsure if this is smart.

But going this explicit, we could implement __str__ on FileLocation and print the file location directly with log.warning()

```
@dataclass
class FileLocation:
  file: str
  line: int

  def __str__(self) -> str:
    return f"{self.file}:{self.line}"

class Operation:
  ...
  fl: FileLocation
```

Reply:
Oh, I didn't think of that. In that case, we should really make the full file path a member of every `Operation`. I guess there won't be millions of operations, so we shouldn't have to worry about this waste of memory.

(In `Book`, we could also keep a dictionary of file paths and IDs, assigning only the corresponding ID to `Operation` -- but I think that's overkill for now.)